apps/docs/content/guides/local-development/testing/overview.mdx

```diff
@@ -0,0 +1,298 @@
+---
+id: 'local-development-testing-overview'
+title: 'Testing Overview'
+description: 'Learn how develop and test database schema, tables, functions, and Row Level Security RLS policies.'
+---
+
+Testing is a critical part of database development, especially when working with features like Row Level Security RLS policies. This guide provides a comprehensive approach to testing your Supabase database.
+
+## Testing approaches
+
+### 1. Database unit testing with pgTAP
+
+[pgTAP](https://pgtap.org) is a unit testing framework for PostgreSQL that allows testing:
+
+- Database structure: tables, columns, constraints
+- Row Level Security RLS policies
+- Functions and procedures
+- Data integrity
+
+#### Testing RLS policies in practice
+
+This example demonstrates setting up and testing RLS policies for a simple todo application:
+
+1. Create a test table with RLS enabled:
+
+```sql
+-- Create a simple todos table
+create table todos (
+  id uuid primary key default gen_random_uuid(),
+  task text not null,
+  user_id uuid references auth.users not null,
+  completed boolean default false
+);
+
+-- Enable RLS
+alter table todos enable row level security;
+
+-- Create a policy
+create policy "Users can only access their own todos"
+on todos for all -- this policy applies to all operations
+to authenticated
+using (auth.uid() = user_id);
+```
+
+2. Set up your testing environment:
+
+```bash
+# Create a new test for our policies using supabase cli
+supabase test new todos_rls.test
+```
+
+3. Write your RLS tests:
+
+```sql
+begin;
+-- install tests utilities
+-- install pgtap extension for testing
+create extension if not exists pgtap with schema extensions;
+-- Start declare we'll have 6 test cases in our test suite
+select plan(6);
+
+-- Setup our testing data
+-- Set up auth.users entries
+insert into auth.users (id, email) values
+    ('123e4567-e89b-12d3-a456-426614174000', 'user1@test.com'),
+    ('987fcdeb-51a2-43d7-9012-345678901234', 'user2@test.com');
+
+-- Create test todos
+insert into public.todos (task, user_id) values
+    ('User 1 Task 1', '123e4567-e89b-12d3-a456-426614174000'),
+    ('User 1 Task 2', '123e4567-e89b-12d3-a456-426614174000'),
+    ('User 2 Task 1', '987fcdeb-51a2-43d7-9012-345678901234');
+
+-- Test 1: Verify RLS is enabled
+select results_eq(
+    $$SELECT relrowsecurity FROM pg_class WHERE relname = 'todos' AND relnamespace = 'public'::regnamespace$$,
+    ARRAY[true],
+    'Row Level Security should be enabled on todos table'
+);
+
+-- Test 2: Verify policy exists
+select policies_are('public', 'todos', ARRAY['Users can only access their own todos']);
+
```

Comment:
```suggestion
```

Is it duplicating code to check for the exact policy? I think executing statements which invoke the policy should be enough.

Reply:
This is to show an example of how you can achieve something similar to

```
-- Test 1: Verify RLS is enabled
select tests.rls_enabled('public', 'todos');

-- Test 2: Verify policy exists
select policies_are('public', 'todos', ARRAY['Users can only access their own todos']);
```

Without the use of the test helpers. Might indeed a very realistic thing to test indeed I'll get rid of it.